catalogue/engage.py

```diff
@@ -17,41 +24,90 @@ def git_query(*args):
     return True
 
 
-def lock(*args):
+def lock(timestamp, input_data, code):
     """
-    TODO: IMPLEMENT!
-
-    - uses ct.create_dictionary() to process inputs
-    - creates .loc file (json structure)
-
-    Parameters:
-        input_data, code, timestamp
+    Create dictionary with hashed inputs and save to .lock file in CATALOGUE_DIR.
+
+    Parameters
+    ----------
+    timestamp : str
+        Datetime.
+    input_data : str
+        Path to input data directory.
+    code : str
+        Path to analysis directory.
     """
-    pass
+    hash_dict = ct.construct_dict(timestamp, input_data, code)
+    if not os.path.exists(CATALOGUE_DIR):
+        os.makedirs(CATALOGUE_DIR)
+    with open(CATALOGUE_LOCK_PATH, "w") as f:
```

Comment:
Should you check that a lock file exists here and raise an exception? Isla's readme proposes that is the behavior we want.